Assets/Scripts/Puzzle/ExitGate.cs

```diff
@@ -0,0 +1,16 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.Events;
+
+public class ExitGate : Gate {
```

Comment:
The only thing that makes this an **exit** gate is the name of the event: `onLevelComplete`. Maybe rename it to something more generic, or move this behaviour into the `Gate` class?

Reply:
Ye, true. Still think it should be it's own gate though. Any good thougths on names?

Reply:
No, not really: GenericGate, TheGate, UsableGate, ActivatableGate, GateGate, DoorGate, Gaterade, Gejt, SIgnalGate, EventGate